projects/spot/grasp.py

```diff
@@ -1,98 +1,225 @@
-        img = imcv2(hand_image_response)
-        pixels = self.find_obj(img=img)
-        print(f" > Grasping object at {pixels}")
+            raise TypeError(f"img is of type {type(img)}, expected is numpy array")
+
+    def grasp(self, hand_image_response, pixels, timeout=10, count=3):
+        # hand_image_response = image_responses_local[0]  # only expecting one image
+        # img = imcv2(hand_image_response)
         k = 0
         while True:
-            success = self.spot.grasp_point_in_image(hand_image_response, 
-                                                pixel_xy=pixels, 
-                                                timeout=timeout,
-                                                top_down_grasp=self.top_grasp, 
-                                                horizontal_grasp=self.hor_grasp
-                                                )
-            if success:
-                self.spot.set_arm_joint_positions(self.stow, travel_time=1.0)
-                break
-            else:
+            # pixels = self.find_obj(img=img)
+            if pixels is not None:
+                print(f" > Grasping object at {pixels}")
+                success = self.spot.grasp_point_in_image(
+                    hand_image_response,
+                    pixel_xy=pixels,
+                    timeout=timeout,
+                    top_down_grasp=self.top_grasp,
+                    horizontal_grasp=self.hor_grasp,
+                )
+                if success:
+                    print(" > Sucess")
+                    self.pick_location = self.spot.get_arm_joint_positions(
+                        as_array=True
+                    )
+                    self.reset_to_stow()
+                    time.sleep(1)
+                    return success
                 k = k + 1
-                print(f" > Could not find object from the labels, tries left: {count - k}")
-            if k == count:
-                print("> Ending trial as target trials reached")
-                retry = input("Would you like to retry? y/n: ")
-                if retry == "y":
-                    #@JAY add a look around script and then replace with gaze
-                    continue
-                else:
+                print(
+                    f" > Could not find object from the labels, tries left: {count - k}"
+                )
+                if k >= count:
+                    print(" > Ending trial as target trials reached")
+                    return success
                     break
-        print('Sucess')
+            else:
+                return None
+                # new_look = self.look
+                # new_look[0] -=np.pi/4
+                # self.spot.set_arm_joint_positions(new_look, travel_time=1)
+                # pixels = self.find_obj(img=imcv2(self.spot.get_image_responses([SpotCamIds.HAND_COLOR])[0]))
         time.sleep(1)
 
+    def update_label(self, new_label: str):
```

Comment:
Create 2 functions, update and add. The current logic is for "add_label" since it's appending